### PR TITLE
Content-Type should not contain multi-values. Closes #986

### LINKED ISSUE #986: Response formatter blows up when headers turn into arrays
https://github.com/restify/node-restify/blob/20f8fa8f0084bfc2e5ff3f981aa24edde32f6758/lib/response.js#L118

I ran into a strange scenario today that I can't understand.

    // Setting the default content-type, assuming we will override it in another middleware
    res.setHeader('content-type', 'text/plain')
    // At this point: this.getHeader('Content-Type') == "text/plain"

    // ... some time later, in another middleware
    res.json(200, { ok: true })
    // At this point: this.getHeader('Content-Type') == ["text/plain", "application/json"]

This is incredibly strange. I don't understand why I would end up with an array of content types. At any rate, the formatter doesn't know how to deal with this as its expecting strings. Yet another fun day with dynamic languages!

### PR DESCRIPTION
**Pull request by jocull**
Fixes https://github.com/restify/node-restify/issues/986

### CODE REVIEW

File: test/server.test.js [hunk 1]
```diff
@@ -1813,6 +1813,23 @@ test('gh-779 set-cookie fields should never have commas', function (t) {
...
+    CLIENT.get('/content-type', function (err, _, res) {
+        t.ifError(err);
+        t.equal(Array.isArray(res.headers['content-type']), false,
+                'content-type header should not be an array');
+        t.equal(res.headers['content-type'], 'bar');
+        t.end();
+    });
+});
+
+
```
Comment (micahr) [inline test/server.test.js]: Can you add another testcase using `res.setHeader`, which should have the same functionality, but just to cover both cases?

**Comment by jocull** [comment]
@micahr Added tests. Got whined at by ESLint so the indentation is a little strange.

File: lib/response.js [hunk 2]
```diff
@@ -178,6 +178,16 @@ Response.prototype.getHeaders = function getHeaders() {
 };
 Response.prototype.headers = Response.prototype.getHeaders;
 
+/**
+ * Headers that cannot be multi-values.
+ * @see #779, don't use comma separated values for set-cookie
+ * @see #986, don't use comma separated values for content-type
+ * @see http://tools.ietf.org/html/rfc6265#section-3
+ */
+var HeaderArrayBlacklist = {
```
Comment (DonutEspresso) [inline lib/response.js]: Since this is a constant, can you cap case it and move it to the top of the file? Under the `/// Globals` section.

**Comment by DonutEspresso** [comment]
Thanks for the PR! LGTM.

**Comment by sean3z** [comment]
I believe the `Set-Cookie` header can legitimately be set multiple times.

File: lib/response.js (hunk 2 above)
Comment (jocull) [inline lib/response.js]: Added!

Comment (DonutEspresso) [inline lib/response.js]: Uppercase please, since we're not yet on ES6  :)

Comment (jocull) [inline lib/response.js]: Sorry, uppercase how? `HEADERARRAYBLACKLIST`, `HEADER_ARRAY_BLACKLIST`, ...

Comment (DonutEspresso) [inline lib/response.js]: `HEADER_ARRAY_BLACKLIST` looks great.

Comment (jocull) [inline lib/response.js]: Done!

File: test/server.test.js [hunk 3]
```diff
@@ -1813,6 +1813,44 @@ test('gh-779 set-cookie fields should never have commas', function (t) {
...
+            t.ifError(err);
+            t.equal(Array.isArray(res.headers['content-type']), false,
+                    'content-type header should not be an array');
+            t.equal(res.headers['content-type'], 'bar');
+            t.end();
+        });
+    });
+
+
+test('gh-986 content-type fields should never have commas'
```
Comment (DonutEspresso) [inline test/server.test.js]: Just noticed this test is identical to the one above - did you mean to add one for set-cookie?

Comment (micahr) [inline test/server.test.js]: He was adding a test that I requested, for the use of `res.header` vs `res.setHeader`

Comment (DonutEspresso) [inline test/server.test.js]: Whoops, I can reads.

**Comment by DonutEspresso** [comment]
Thanks @jocull!

**Pull request merged**